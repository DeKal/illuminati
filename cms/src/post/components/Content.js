import React from 'react'
import PropTypes from 'prop-types'
import CKEditor from '@ckeditor/ckeditor5-react'
import ClassicEditor from '@ckeditor/ckeditor5-build-classic'

const Content = ({ content, editPostContent }) => {
  return (
    <CKEditor
      editor={ClassicEditor}
      data={content}
      onChange={(_, editor) => {
        const data = editor.getData()
        editPostContent(data)
      }}
    />
  )
}

export default Content

Content.propTypes = {
  isEdited: PropTypes.bool,
  content: PropTypes.string,
  editPostContent: PropTypes.func
}
