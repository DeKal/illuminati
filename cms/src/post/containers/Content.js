import { connect } from 'react-redux'
import { getEditedContent } from 'post/state/selectors'
import { editPostContent } from 'post/state/actions'
import Content from 'post/components/Content'

const mapStateToProps = state => {
  const editedContent = getEditedContent(state)
  return {
    content: editedContent || ''
  }
}

const mapDispatchToProps = {
  editPostContent
}

export default connect(mapStateToProps, mapDispatchToProps)(Content)
