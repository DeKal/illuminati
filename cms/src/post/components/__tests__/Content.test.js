import * as React from 'react'
import { shallow } from 'enzyme'
import Content from '../Content'

describe('Content', () => {
  it('should render Content correctly', function() {
    const wrap = shallow(<Content content="content" />)
    expect(wrap).toMatchSnapshot()
  })
})
