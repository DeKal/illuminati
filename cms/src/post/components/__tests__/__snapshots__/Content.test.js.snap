// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render Content correctly 1`] = `
<a
  config={Object {}}
  data="content"
  editor={[Function]}
  onChange={[Function]}
/>
`;
